MyLibrary/Library/Controllers/GenreController.cs

```diff
@@ -0,0 +1,155 @@
+﻿// <copyright file="GenreController.cs" company="Peretiatko Anastasiia">
+// Copyright (c) Peretiatko Anastasiia. All rights reserved.
+// </copyright>
+
+using System;
+using AutoMapper;
+using BusinessLogic.LibraryModel;
+using BusinessLogic.LibraryService;
+using Microsoft.AspNetCore.Mvc;
+using MyWebLibrary.ViewModel;
+
+namespace MyWebLibrary.Controllers
+{
+    /// <summary>
+    /// Controller for dealing with genres
+    /// </summary>
+    /// <seealso cref="Microsoft.AspNetCore.Mvc.ControllerBase" />
+    [Route("api/[controller]")]
+    [ApiController]
+    public class GenreController : ControllerBase
+    {
+        /// <summary>
+        /// The library service
+        /// </summary>
+        private IGenreService libraryService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GenreController"/> class
+        /// </summary>
+        /// <param name="libraryService">The library service</param>
+        public GenreController(IGenreService libraryService)
+        {
+            this.libraryService = libraryService;
+        }
+
+        // GET: api/genre
+        /// <summary>
+        /// Gets all genres from the Library
+        /// </summary>
+        /// <returns>
+        /// HTTP status code Ok() with the list of genres
+        /// or HTTP status code NoContent() if there are any genres in library
+        /// </returns>
+        [HttpGet]
+        public IActionResult GetAllGenres()
```

Comment:
Here as well, is more elegantly will fil method overloading GetGenres()/GetGenres(int id) for this and next method.